src/peer_list.rs

```
@@ -169,6 +169,50 @@ impl<S: SecretId> PeerList<S> {
         }
     }
 
+    /// Initialise the membership list of `peer_id` with the given `other_peer_ids`.
+    pub fn initialise_membership_list<I>(&mut self, peer_id: &S::PublicId, other_peer_ids: I)
+    where
+        I: IntoIterator<Item = S::PublicId>,
+    {
+        // Do not populate our membership list, as it would be redundant.
+        if *peer_id == *self.our_id.public_id() {
+            return;
+        }
+
+        if let Some(peer) = self.peers.get_mut(peer_id) {
```

`else`, shouldn't we return an error? (To make this function harder to misuse without noticing)